zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -26,57 +28,77 @@
     implements ColumnFamily<KeyType, ValueType> {
 
   private final ZeebeTransactionDb<ColumnFamilyNames> transactionDb;
-  private final long handle;
-
-  private final DbContext context;
+  private final TransactionContext context;
 
-  private final DbCompositeKey<DbLong, KeyType> compositeKey;
   private final ValueType valueInstance;
   private final KeyType keyInstance;
-  private final DbLong columnFamilyKey;
+  private final ColumnFamilyNames columnFamilyName;
+  private final ColumnFamilyContext columnFamilyContext;
 
   TransactionalColumnFamily(
       final ZeebeTransactionDb<ColumnFamilyNames> transactionDb,
       final ColumnFamilyNames columnFamily,
-      final DbContext context,
+      final TransactionContext context,
       final KeyType keyInstance,
       final ValueType valueInstance) {
     this.transactionDb = transactionDb;
-    handle = this.transactionDb.getColumnFamilyHandle();
     this.context = context;
     this.keyInstance = keyInstance;
     this.valueInstance = valueInstance;
-    columnFamilyKey = new DbLong();
-    columnFamilyKey.wrapLong(columnFamily.ordinal());
-    compositeKey = new DbCompositeKey<>(columnFamilyKey, keyInstance);
+    columnFamilyName = columnFamily;
+    columnFamilyContext = new ColumnFamilyContext();
   }
 
-  @Override
-  public void put(final KeyType key, final ValueType value) {
-    put(context, key, value);
+  private void ensureInOpenTransaction(
+      final TransactionContext context, final TransactionConsumer operation) {
+    context.runInTransaction(
+        () -> operation.run((ZeebeTransaction) context.getCurrentTransaction()));
```

Comment:
Why is this cast needed? Can't we just use the interface `ZeebeDbTransaction`?

Reply:
Yeah because currently the put,get stuff is in the Transaction impl, but I might get around that. Let me try something.

Reply:
I will not change it now. In this way it is easier and we are inside the implementation which means we know exactly the class. I thought we can move the put to the column family class but it needs the reference to the transaction objects, which is not easier without exposing it via the interface which I dont want.